src/pages/patch/patchModal/PatchModal.tsx

```diff
@@ -0,0 +1,228 @@
+import React, { useState } from "react";
+import { Modal } from "antd";
+import { useQuery, useMutation } from "@apollo/react-hooks";
+import styled from "@emotion/styled";
+import Button from "@leafygreen-ui/button";
+import Checkbox from "@leafygreen-ui/checkbox";
+import { Body, H2 } from "@leafygreen-ui/typography";
+import { uiColors } from "@leafygreen-ui/palette";
+import { useParams } from "react-router-dom";
+import { TreeSelect } from "components/TreeSelect";
+import { GET_PATCH_BUILD_VARIANTS } from "gql/queries/get-patch-build-variants";
+import { RESTART_PATCH } from "gql/mutations/restart-patch";
+import get from "lodash/get";
+import {
+  PatchBuildVariantsQuery,
+  PatchBuildVariantsQueryVariables,
+  RestartPatchMutation,
+  RestartPatchMutationVariables,
+} from "gql/generated/types";
+import { usePatchStatusSelect } from "hooks";
+import { useBannerDispatchContext } from "context/banners";
+import { TaskStatus } from "types/task";
+import { PatchBuildVariantAccordian } from "./PatchBuildVariantAccordian";
+
+const { gray } = uiColors;
+
+interface PatchModalProps {
+  visible: boolean;
+  onOk: () => void;
+  onCancel: () => void;
+}
+export const PatchModal: React.FC<PatchModalProps> = ({
+  visible,
+  onOk,
+  onCancel,
+}) => {
+  const { successBanner, errorBanner, clearAll } = useBannerDispatchContext();
+  const { id } = useParams<{ id: string }>();
+  const [abortInProgressTasks, setAbortInProgressTasks] = useState(false);
+  const [restartPatch, { loading: mutationLoading }] = useMutation<
+    RestartPatchMutation,
+    RestartPatchMutationVariables
+  >(RESTART_PATCH, {
+    onCompleted: () => {
+      onOk();
+      successBanner(`Successfully restarted patch: ${id}`);
```

Comment:
we do not use the patch id anywhere in the app to identify the patch (other than the url). therefore i do not think we should use it here in the success message. We could use the patch number, the description, or none of those at all and just say `success!`